src/PolyhedralGeometry/Polyhedron/properties.jl

```diff
@@ -728,6 +728,64 @@ julia> codim(P)
 """
 codim(P::Polyhedron) = ambient_dim(P)-dim(P)
 
+@doc raw"""
+    facet_sizes(P::Polyhedron{T})
+
+Number of vertices in each facet. 
+
+#Example
+```jldoctest
+julia> p = johnson_solid(3) 
+Polyhedron in ambient dimension 3 with EmbeddedElem{nf_elem} type coefficients
+
+julia> facet_sizes(p) 
+8-element Vector{Int64}:
+ 3
+ 4
+ 3
+ 3
+ 4
+ 6
+ 3
+ 4
+```
+"""
+function facet_sizes(P::Polyhedron{T}) where T<:scalar_types
+  is_bounded(P) && return Vector{Int}(pm_object(P).FACET_SIZES)
+  im = IncidenceMatrix(pm_object(P).VERTICES_IN_FACETS)
+  nrows = size(im)[1]
+  #get the vertices that are really vertices and the facets that are really facets
+  ff = convert(Set{Int}, Polymake.to_one_based_indexing(P.pm_polytope.FAR_FACE))
+  i = _facet_at_infinity(pm_object(P))
+  rows = Vector{Int}(1:nrows)
+  i <= nrows && deleteat!(rows,i) #the face at infinity is not always a facet, so does not always need to be removed
+  return [length(Base.setdiff(row(im,i), ff)) for i in 1:length(rows)]
```

Comment:
```suggestion
  im = vertex_indices(facets(P))
  return [length(row(im, i)) for i in 1:nrows(im)]
```

`vertex_indices` should take care of ignoring rays. `facets` should ignore the far facet if necessary. And I don't think we need the `is_bounded` special case.